pkg/config/config.go

```diff
@@ -58,88 +62,6 @@ var (
 	DefaultImage = "gcr.io/heptio-images/sonobuoy:" + buildinfo.Version
 )
 
-///////////////////////////////////////////////////////
-// Note: The described resources are a 1:1 match
-// with kubectl UX for consistent user experience.
-// xref: https://kubernetes.io/docs/api-reference/v1.11/
-///////////////////////////////////////////////////////
-
-// ClusterResources is the list of API resources that are scoped to the entire
-// cluster (ie. not to any particular namespace)
-var ClusterResources = []string{
-	"CertificateSigningRequests",
-	"ClusterRoleBindings",
-	"ClusterRoles",
-	"ComponentStatuses",
-	"CustomResourceDefinitions",
-	"Nodes",
-	"PersistentVolumes",
-	"PodSecurityPolicies",
-	"ServerGroups",
-	"ServerVersion",
-	"StorageClasses",
-}
-
-// NamespacedResources is the list of API resources that are scoped to a
-// kubernetes namespace.
-var NamespacedResources = []string{
-	"ConfigMaps",
-	"ControllerRevisions",
-	"CronJobs",
-	"DaemonSets",
-	"Deployments",
-	"Endpoints",
-	"Events",
-	"HorizontalPodAutoscalers",
-	"Ingresses",
-	"Jobs",
-	"LimitRanges",
-	"NetworkPolicies",
-	"PersistentVolumeClaims",
-	"PodDisruptionBudgets",
-	"PodLogs",
-	"PodTemplates",
-	"Pods",
-	"ReplicaSets",
-	"ReplicationControllers",
-	"ResourceQuotas",
-	"RoleBindings",
-	"Roles",
-	"ServiceAccounts",
-	"Services",
-	"StatefulSets",
-}
-
-// NamespacedDefaults is the default-list of API resources to gather
-// TODO (tstclair): Clean up defaulting, this is temporary
-var NamespacedDefaults = []string{
-	"ConfigMaps",
-	"ControllerRevisions",
-	"CronJobs",
-	"DaemonSets",
-	"Deployments",
-	"Endpoints",
-	//"Events",
-	//"HorizontalPodAutoscalers",
-	"Ingresses",
-	"Jobs",
-	"LimitRanges",
-	"NetworkPolicies",
-	"PersistentVolumeClaims",
-	"PodDisruptionBudgets",
-	//"PodLogs",
```

Comment:
Need to review the pod logs logic right now. I think as I coded it, it _only_ captures the logs when in the heptio-sonobuoy namespace. Similar question about events and HPA, as coded now, we always get those in the default crawl for stuff.

So the question becomes, how do the users get pod logs in all namespaces if they wanted? They used to add it to the list of resources, but the default list is empty (indicating to dynamically get everything).

A long, manually maintained list like this is awkward for users and devs alike. I think it makes sense to be more opinionated even if it makes a slightly breaking change in behavior. If we do that, I'd just want to get slightly more data instead of erring out and getting no data.

This manual list had the draw back that you had no clear idea what you could add, but if you figured it out you would add the one item and it would work. With allowing the empty value to be 'all things' it makes it harder to specify things like "all but one" or "default and one more thing".

Thinking...

Reply:
At the moment I've feel like I've come to the right conclusion: I was being too hasty and trying to remove too much here for the sake of "simplicity". Removing all these defaults is in a sense simple, but changing the model completely like that does either break specific tasks/flows or leaves some questions to be decided suddenly about what to do/how to do it.

Instead, I'm going to opt for the less dramatic change:
 - continue to have a list of default options we query for
 - update the names in the list so I don't have to worry about all the casing concerns; its not a fun little section of change to double check spelling of 30-50 constants but it is what it is
 - if someone still wants to really query _everything_ they can just remove the list of resources so it is nil and implicitly get _everything_.

This will enable me to maintain everything about the current flow I think: in the "sonobuoy" namespace we can remove the same values as we do now, if someone wants to add podlogs or secrets its as easy as adding something to the list.

When we extract queries to their own plugin we can modify more of this behavior in a more natural way.